syntax = "proto2";

import "meteo.proto";

package meteo.settings;


enum NodeType {
  kGGMCenter               = 0; //!<Главный гидрометеорологический центр МО РФ
  kRMCenter                = 1; //!<Радиометеорологический (Р) центр
  kGMCenter                = 2; //!<гидрометеорологический (Г) центр
  kMCenter                 = 3; //!<метеорологический (М) центр
  kGMCBuroStation          = 4; //!<Гидрометеорологическая станция, метеобюро (А - армии)
  kOGMrotaStation          = 5; //!<Отдельная гидрометеорологическая рота
  kMEtVzvodStation         = 6; //!<Метеорологический взвод
//!<Метеорологические станции:
  kStation                 = 7; //!<  1 – общее обозначение;
  kGMStation               = 8; //!<  2 –гидрометеорологическая базовая;
  kAvtoRMStation           = 9; //!<  3 – автомобильная радиотелеметрическая;
  kMobileStation           = 10; //!<  4 – подвижная (маневренная);
  kAutoStation             = 11; //!<  5 – автоматическая радиометеорологическая
//!<Посты:
  kMeteoPost               = 12; //!<  1 – метеорологический;
  kHydroStation            = 13; //!<  2 – гидрологический
  kARM                     = 14; //!<Программно-технический комплекс гидрометеорологического обеспечения
  kAPPI                    = 15; //!<Автономный пункт приема спутниковой информации
  kMRL                     = 16; //!<Метеорологический радиолокатор
  kCrypto                  = 17; //!<Метеорологическая кодировочная аппаратура
  kVVrotaGydromet          = 18; //!<Военно-восстановительная рота Росгидромета
  kKVNYStation             = 19; //!<Координирующие и подчиненные межрегиональные территориальные управления Росгидромета
  kMRLGydromet             = 20; //!<Метеорологический радиолокатор Росгидромета
}


message Location {
  optional string index                   = 1;  //!< идентификатор пункта
  optional string name                    = 2;  //!< название пункта
  optional string unit_number             = 3;  //!< номер ВЧ
  optional meteo.GeoPointPb coord = 4;  //!< координаты
  optional string fio_cmdr                = 5;  //!< фио командира ВЧ
  optional int32 rank_cmdr                = 6;  //!< звание командира ВЧ
  optional string wmo_id                  = 7;  //!< ВМО-идентификатор
  optional string hms_id                  = 8;  //!< ГМС-идентификатор
  repeated string ttaa                    = 9;  //!< типы данных предоставляемых станцией
  optional bool   tlg_distributor         = 10 [ default = false ]; //!< признак, указывающий, что станция обеспечивыет рассылку телеграмм
  optional NodeType station_type = 20 [ default = kStation ];  //!< тип станции
}
